packages/dataviews/src/view-list.tsx

```diff
@@ -335,10 +335,8 @@ export default function ViewList< Item extends AnyItem >(
 			)
 	);
 
-	const onSelect = useCallback(
-		( item: Item ) => onSelectionChange( [ item ] ),
-		[ onSelectionChange ]
-	);
+	const onSelect = ( item: Item ) =>
+		onSelectionChange( [ getItemId( item ) ] );
```

Comment:
Note that `useCallback` is not needed here, it's just used in a React event callback down the line.